### PR TITLE
switch to struct init for Renderer and Texture creation, like wgpu-rs

### PR DESCRIPTION
**Pull request by benmkw**
ref https://github.com/Yatekii/imgui-wgpu-rs/pull/33

happy to make naming changes or other stuff if recommended :)

### CODE REVIEW

**Review by Yatekii — APPROVED** [review]
Thanks for the works!

Looks pretty good to me, what do you think, @cwfitzgerald ? :)

Hmm we should also fix all the clippy lints some time, I guess I'll have to do that ^^

File: src/lib.rs [hunk 1]
```diff
@@ -283,17 +230,21 @@ impl RendererConfig<'_, '_> {
...
         }
     }
+}
 
+impl Default for RendererConfig<'_, '_> {
     /// Create a new renderer config with precompiled default shaders outputting linear color.
     ///
     /// If you write to a Bgra8UnormSrgb framebuffer, this is what you want.
-    pub fn new() -> RendererConfig<'static, 'static> {
+    fn default() -> Self {
```
Comment (cwfitzgerald) [inline src/lib.rs:239]: We shouldn't get rid of the new impl, it's generally good to have a new impl default just directly calls.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Got some nits, but beyond that looks good

File: src/lib.rs (hunk 1 above)
Comment (benmkw) [reply src/lib.rs:239]: Yeah I was not sure about that but makes sense. I also replaced some struct names where possible with Self as that seems to be the way clippy likes it.

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**